AElf.Contract.Resource.Tests/ResourceContractTest.cs

```diff
@@ -99,9 +101,70 @@ public async Task Verify_FeeReceiverContract_Information()
         }
 
         [Fact]
-        public async Task Verify_Resource_AddressInfo()
+        public async Task FeeReceiver_WithDraw_WithoutPermission()
+        {
+            await Initialize_Resource();
+
+            Tester.SetCallOwner(CryptoHelpers.GenerateKeyPair());
+            var withdrawResult = await Tester.ExecuteContractWithMiningAsync(FeeReceiverContractAddress, "Withdraw",
+                100);
+            withdrawResult.Status.ShouldBe(TransactionResultStatus.Failed);
+            var retValue = withdrawResult.RetVal.ToStringUtf8();
+            retValue.Contains("Only foundation can withdraw token.").ShouldBeTrue();
+        }
+
+        [Fact]
+        public async Task FeeReceiver_WithDraw_OverToken()
+        {
+            await Initialize_Resource();
+
+            Tester.SetCallOwner(FoundationKeyPair);
+            var withdrawResult = await Tester.ExecuteContractWithMiningAsync(FeeReceiverContractAddress, "Withdraw",
+                100);
+            withdrawResult.Status.ShouldBe(TransactionResultStatus.Failed);
+            var retValue = withdrawResult.RetVal.ToStringUtf8();
+            retValue.Contains("Too much to withdraw.").ShouldBeTrue();
+        }
+
+        [Fact]
+        public async Task FeeReceiver_WithDraw_NormalCase()
+        {
+            await Initialize_Resource();
+
+            Tester.SetCallOwner(FoundationKeyPair);
+            var withdrawResult = await Tester.ExecuteContractWithMiningAsync(FeeReceiverContractAddress, "Withdraw",
+                0);
+            withdrawResult.Status.ShouldBe(TransactionResultStatus.Mined);
+        }
+
+        [Fact]
+        public async Task FeeReceiver_WithDraw_all()
         {
-            await Initize_Resource();
+            await Initialize_Resource();
+
+            Tester.SetCallOwner(FoundationKeyPair);
+            var withdrawResult = await Tester.ExecuteContractWithMiningAsync(FeeReceiverContractAddress, "WithdrawAll");
+            withdrawResult.Status.ShouldBe(TransactionResultStatus.Mined);
+        }
+
+        [Fact(Skip = "Anyone can burn foundation token until token value is smaller than foundation owed fee.")]
+        public async Task FeeReceiver_Burn()
+        {
+            await Initialize_Resource();
```

Comment:
move to constructor?

Reply:
xunit also provider some method to start before a method, and clean after some method